discovery/manager_test.go

```diff
@@ -949,6 +951,91 @@ scrape_configs:
 	}
 }
 
+func TestGaugeFailedConfigs(t *testing.T) {
+	var (
+		fcGauge prometheus.Gauge
+		err     error
+	)
+
+	cfgOneText := `
+scrape_configs:
+- job_name: prometheus
+  consul_sd_configs:
+  - server: "foo:8500"
+    tls_config:
+      cert_file: "/tmp/non_existent"
+  - server: "bar:8500"
+    tls_config:
+      cert_file: "/tmp/non_existent"
+  - server: "foo2:8500"
+    tls_config:
+      cert_file: "/tmp/non_existent"
+`
+	cfgOne := &config.Config{}
+
+	err = yaml.UnmarshalStrict([]byte(cfgOneText), cfgOne)
+	if err != nil {
+		t.Fatalf("Unable to load YAML config cfgOne: %s", err)
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	discoveryManager := NewManager(ctx, log.NewNopLogger())
+	discoveryManager.updatert = 100 * time.Millisecond
+	go discoveryManager.Run()
+
+	c := make(map[string]sd_config.ServiceDiscoveryConfig)
+	for _, v := range cfgOne.ScrapeConfigs {
+		c[v.JobName] = v.ServiceDiscoveryConfig
+	}
+
+	discoveryManager.ApplyConfig(c)
+	<-discoveryManager.SyncCh()
+
+	metricOne := &dto.Metric{}
+	fcGauge, err = failedConfigs.GetMetricWithLabelValues(discoveryManager.name)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	fcGauge.Write(metricOne)
+
+	failedCount := metricOne.GetGauge().GetValue()
```

Comment:
There is https://godoc.org/github.com/prometheus/client_golang/prometheus/testutil#ToFloat64 to save you the boilerplate code here.

Reply:
Good to know about it. But, I'd like to have another PR for this whole test file `manager_test.go`, what do you think ?

Reply:
If you mean another PR to refactor `manager_test.go` to make use of testutil, I think that's reasonable.

Reply:
> If you mean another PR to refactor `manager_test.go` to make use of testutil, I think that's reasonable.

Exactly, I will ask you for a review once I create another PR to refactor it.